File: client/src/app/models/Order.ts
 export interface Order  {
  id: number
  buyerId: string
  shippingAddress: ShippingAddress
  date: string
  orderItems: OrderItem[]
  subtotal: number
  deliveryFee: number
  orderStatus: string
  total: number
}

export interface ShippingAddress {
  fullName: string
  address1: string
  address2: string
  city: string
  state: string
  zip: string
  country: string
}

export interface OrderItem {
  productId: number
  name: string
  pictureUrl: string
  price: number
  quantity: number
}

File: API/Entities/orderAggregates/Order.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities.orderAggregates
{
    public class Order
    {
        public int Id { get; set; }
        public string BuyerId { get; set; }
        [Required]
        public shippingAddress ShippingAddress { get; set; }
        public DateTime Date { get; set; } = DateTime.UtcNow;

        public List<OrderItems> OrderItems { get; set; }

        public long Subtotal { get; set; }
        public long DeliveryFee { get; set; }

        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;

        public string PaymentIntentId { get; set; }

        public long GetTotal() => Subtotal + DeliveryFee;


    }
}